### DISCUSSION TITLE
haproxy-status3: Non-sense check

### DISCUSSION THREAD

**Discussion by 10RUPTiV** [Linuxfabrik/monitoring-plugins#592] [General]
Hi,

The data gathering some haproxy csv is having some part that need to be excluded from the check!

![image](https://user-images.githubusercontent.com/20915668/164525665-163ac36a-213e-47ce-af6a-e2db75d9d4f9.png)

The SERVERNAME " BACKEND " will always be /1.0 , that mean it will always generate a CRITICAL.

We spoke with Haproxy team (we have a business contract with them) and they told us, there's no way in the config to increase the /1.0 on for this one.

Maybe it should be remove from the "check" ?

**Comment by markuslf** [comment]
I will have a look into it. 👍

**Comment by markuslf** [comment]
It seems that you are referring to the `Sessions` column, and you get a CRIT, because 2 out of 1 possible sessions are used (which is weird).

I had a look at the plugin output in one of our datacenters. There, the output looks like this:
![image](https://user-images.githubusercontent.com/31855393/174315184-f08c4170-7377-4e85-9212-b3706cb5ae53.png)

So in BACKEND, our HAproxy supports up to 300 connections. In this special case, we are running HAProxy version 2.4.17, and check plugin version 2022021602. What do you use on your side?

**Comment by 10RUPTiV** [comment]
Markus,

We are using HAproxy EE 2.4r1 and the latest plugin...

But in our backend config, we have not set any limit... that's probably why we have a limit of "1"

**Comment by 10RUPTiV** [comment]
Just did a little test and I can confirm that having no maxconn in the backend setting, equal "unlimited" until we reach the GLOBAL maxconn

**Comment by markuslf** [comment]
Closing this issue. If this check warns about sessions, this is a HAProxy configuration issue. Unsetting or setting `maxconn` to appropriate values helps (as intended and expected).

**Comment by markuslf** [comment]
This is the haproxy.conf related to my screenshot above (relevant parts only):

```
global
    # LUA-module implementing CORS
    lua-load /etc/haproxy/cors.lua
    log         127.0.0.1 local2 debug
    maxconn     4000
    ...


defaults
    log                     global
    maxconn                 3000
    mode                    http
    option                  dontlognull
    option                  redispatch
    ...


frontend 898e875b-47c6-49f4-8858-12d025847b9d
    ....


listen c208297e-326d-471f-a961-2085ec8de1d8
    ...

listen 48a8cc65-fe61-42f2-badf-c8a99a4f22f2
    ...


backend 06cbdfd2-0fd2-46d2-ad4a-4d7fdee29b18
    ...

backend 402bc2b5-199d-4395-adfe-9697f0ae45b0
    ...

backend 97546241-cf82-47b5-aec5-52f22e0768b8
    ...

backend 6f5adba8-b7ee-4e64-a256-5c9c82b087df
    ...

backend f5729c7d-a93a-4ed8-b30b-38f59e5bd618
    ...

backend 84a3a15e-6646-46f1-87e9-2e0c30d43d6c
    ...

backend 828b9168-d4d3-477c-af5d-37d8f3f3d7b1
    ...

backend 8bed5c9c-3af4-433e-a822-a0e2555fa064
    ...

backend 1993ff94-4d8f-46b9-8017-943f34e23269
    ...

backend 3ba4792a-d5eb-4536-9291-243a5d5b833a
    ...

backend f019651d-1067-4383-9f1b-4be2729c238a
    ...

backend 98417bf7-de22-43fd-a565-5cda16e1073a
    ...

backend 127d222c-60b5-4d65-9226-e730c7fd9adb
    ...

backend 6ef6a595-be24-47c3-b53e-5fe39085724b
    ...

backend fbf44e2e-1a47-407a-b2c5-9d6cb0b204c2
    ...

```
